veps/sig-compute/45-in-place-hotplug/vep.md

```diff
@@ -0,0 +1,125 @@
+* Support hotplug for non-migratable VMs.
+
+## Non Goals
+
+* The user to explicitly decide whether to use in-place pod resize or live-migration (as migration doesn't really makes sense anymore).
+
+## Definition of Users
+
+* VM owners.
+* Admins / namespace owners.
+
+## User Stories
+
+* As a user, I want to hotplug CPU and memory to my VM without having to live-migrate it.
+* As an admin, I want to save cluster resources and improve performance by avoiding live-migrations.
+* As a namespace owner with a ResourceQuota, I want to be able to hotplug CPU and memory to my VMs without having to worry about the quota being exceeded.
+
+## Repos
+
+kubevirt/kubevirt
+
+## Design
+
+Currently, whenever a VM is hotplugged, virt-controller updates a condition that triggers the workload updater controller
+which leads to a live-migration of the VM.
+
+With this VEP is implemented, the controller would simply change the pod's resources and wait for them to be applied by kubelet.
+In turn, the workload updater controller would avoid live-migrating on this situation.
+
+## API Examples
+
+No API changes are expected.
+
+## Alternatives
+
+An alternative to completely dropping the live-migration update method is to keep it as a secondary option that needs
+to be explicitly enabled by the user. This could potentially help in situations where the pod cannot increase its resources
+due to node constraints or other reasons.
+
+## Risk and Limitations
+
+Currently, the Kubernetes in-place pod resize feature does not support dedicated CPUs
+[[1](https://github.com/kubernetes/enhancements/tree/61abddca34caac56d22b7db48734b7040dc68b43/keps/sig-node/1287-in-place-update-pod-resources#static-cpu--memory-policy)].:
+>Resizing pods with static CPU & memory policy configured is out-of-scope for the beta release of in-place resize. If a pod is a guaranteed QOS on a node with a static CPU or memory policy configured, then the resize will be marked as infeasible.
+
+This issue is currently being addressed and will probably be resolved at the GA phase.
+In the worst case scenario, we can fall-back to the migration method for such VMs.
+
+## Scalability
+
+This should improve scalability dramatically as it reduces the number of live-migrations that need to be performed during
+hotplugs.
+
+## Update/Rollback Compatibility
+
+No update/rollback compatibility issues are expected.
+
+## Functional Testing Approach
+
+Hotplug tests should be updated to test the in-place pod resize feature.
+
+## Implementation Phases
+
+<!--
+How/if this design will get broken up into multiple phases)
+-->
+
+## Feature lifecycle Phases
+
+<!--
+How and when will the feature progress through the Alpha, Beta and GA lifecycle phases
+
+Refer to https://github.com/kubevirt/community/blob/main/design-proposals/feature-lifecycle.md#releases for more details
+-->
+
+### Alpha
+
+- [ ] Implement in-place pod resize for CPU and memory hotplugs.
+
+### Beta
+- [ ] Turn this feature on by default.
+
```

Comment:
I think we should add GA Kubernetes In-Place resize as a requirement

Reply:
My opinion is that beta features should (in most cases) be enabled by default, this is also Kubernetes' policy and I think it makes a lot of sense.

There's a related discussion about about this in https://github.com/kubevirt/kubevirt/pull/14427, which aims to enable a path forward for beta features being on by default.

Reply:
Yes, I understand that Beta features should be enabled by default. This is why I think it would be better to wait until this feature gate is GA in Kubernetes before making it Beta in KubeVirt. I'm not sure we should have default behavior that depends on Kubernetes Beta/Alpha feature gates. We have a similar requirement for the ImageVolume feature gate.

Reply:
The general rule is that our feature gate will be 1 graduation phase below Kubernetes.
ImageVolumes are Beta on Kubernetes, therefore we allow it to become Alpha in Kubevirt.
Same goes for Beta and Alpha with the pod resize FG.